napi.h

```diff
@@ -314,16 +322,20 @@ using MaybeOrValue = T;
 /// corresponds to an Isolate.
 class BasicEnv {
  private:
-  node_api_nogc_env _env;
+  node_addon_api_basic_env _env;
 #if NAPI_VERSION > 5
   template <typename T>
   static void DefaultFini(Env, T* data);
   template <typename DataType, typename HintType>
   static void DefaultFiniWithHint(Env, DataType* data, HintType* hint);
 #endif  // NAPI_VERSION > 5
  public:
-  BasicEnv(node_api_nogc_env env);
+  BasicEnv(node_addon_api_basic_env env);
+#ifdef NODE_API_EXPERIMENTAL_HAS_POST_FINALIZER
```

Comment:
In the node-addon-api meeting on 4 Oct, we discussed that it should be possible just to have a operator to `node_addon_api_basic_env` and `node_addon_api_basic_finalize`, as the compiler will allow conversion from `BasicEnv` -> `node_addon_api_basic_env` -> `node_api_nogc_env` or `napi_env` depending on options allowed.

Reply:
Updated, ptal again, thanks